Tasks/AzureRmWebAppDeployment/KuduUtility.ts

```diff
@@ -0,0 +1,89 @@
+import Q = require('q');
+import tl = require('vsts-task-lib/task');
+import path = require("path");
+import fs = require("fs");
+var request = require ('request');
+var archiver = require('archiver');
+
+
+/**
+ *  Deploys a zip based webapp package.
+ * 
+ *  @param  webAppPackage                  Zip file or folder for deployment
+ *  @param  virtualApplicationMappings     Mapping to get physical path for deployment
+ *  @param  publishingProfile              publish profile provides destination details for deployment
+ *  @param  virtualApplication             (Optional) Virtual application name 
+ */
+export async function deployWebAppPackage(webAppPackage: string, virtualApplicationMappings, publishingProfile,
+                             virtualApplication: string) {
+
+    var deferred = Q.defer<any>();	
+    // construct URL depending on virtualApplication or root of webapplication 
+    var physicalDeploymentPath = "/site/wwwroot";
+    var virtualPath = "/";
+    if (virtualApplication) {
+		virtualPath = "/" + virtualApplication;
+    }
+
+    for( var index in virtualApplicationMappings ){
+        var mapping = virtualApplicationMappings[index];
+        if( mapping.virtualPath == virtualPath){
+            physicalDeploymentPath = mapping.physicalPath;
+            break;
+        }
+    }
```

Comment:
Why is resolving right virtual path a responsibility of this funciton. Why can't caller resolve and provide right path and this function only focuses on deployment?